### PR TITLE
Extract PageTemplate components

### PR DESCRIPTION
**Pull request by dgalarza**
This extracts a CSS Grid for laying out content within a page.

 The following areas are supported:
"breadcrumbs"
"title"
"action-bar"
"content"
"footer"

This consolidates a few "layout" components into the PageTemplate as well such as:

* FooterActions / StickyFooter have been removed in favor of the PageFooter component
* PageTitle has been removed from typography in favor of PageHeader component which will support additional content within the header in the near future.
* ActionBar has been moved into `PageActionBar`

Once published, these packages will be yanked from npm.

Also fixes some issues with the grid system in `AppTemplate`

**Basic page template**
Screen Shot 2023-01-20 at 12 08 37 PM

**Page with action bar**
Screen Shot 2023-01-20 at 12 08 41 PM

**Page with footer**
Screen Shot 2023-01-20 at 12 08 44 PM

### CODE REVIEW

File: packages/components/page_template/src/page_header.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+type PageHeaderProps = OwnProps & Omit<FlexProps, "gridArea">;
+
+export const PageHeader: React.FC<PageHeaderProps> = ({
+  title,
+  ...props
+}): React.ReactElement => {
+  return (
+    <Flex
+      as="header"
+      data-testid="page-header"
```
Comment (acostanzo) [inline packages/components/page_template/src/page_header.tsx:17]:
```suggestion
      gridArea="header"
```

File: packages/components/page_template/src/page_header.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      mb="l"
+      ml="page.gutterX"
+      mt="xxxl"
+      {...props}
+    >
+      <Heading as="h1" gridArea="page-title" size="xl">
+        {title}
+      </Heading>
```
Comment (acostanzo) [inline packages/components/page_template/src/page_header.tsx:25]: Are you waiting to add children as an option until it's needed?

**Review by acostanzo — APPROVED** [review]
Couple minor notes but LGTM!

File: packages/components/page_template/src/page_header.tsx (hunk 2 above)
Comment (dgalarza) [reply packages/components/page_template/src/page_header.tsx:25]: That's correct, once I get to that part of infirmary I'll tackle it.

File: packages/components/page_template/src/page_header.tsx (hunk 1 above)
Comment (dgalarza) [reply packages/components/page_template/src/page_header.tsx:17]: :doh:

**Pull request merged**